tools/targets/PSOC6.py

```diff
@@ -0,0 +1,167 @@
+#
+# Copyright (c) 2017-2018 Future Electronics
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import platform
+import subprocess
+import errno
+from array import array
+from distutils.spawn import find_executable
+from shutil import copyfile
+from intelhex import IntelHex
+from intelhex.compat import asbytes
+
+from ..config import ConfigException
+
+
+class HookError(Exception):
```

Comment:
Please import this instead of defining your own. This makes it quite a bit more difficult to catch your `HookError`.